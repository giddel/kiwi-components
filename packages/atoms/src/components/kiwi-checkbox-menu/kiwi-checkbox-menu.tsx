import { Component, ComponentInterface, h, JSX } from '@stencil/core';

@Component({
  tag: 'kiwi-checkbox-menu',
  styleUrl: 'kiwi-checkbox-menu.css',
  shadow: false,
})
export class KiwiCheckboxMenu implements ComponentInterface {
  render(): JSX.Element {
    return (
      <ul class="checkbox-menu mb-0">
        <slot />
      </ul>
    );
  }
}
