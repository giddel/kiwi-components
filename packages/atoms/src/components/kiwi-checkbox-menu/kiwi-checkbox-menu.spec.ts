import { newSpecPage } from '@stencil/core/testing';
import { KiwiCheckboxMenu } from './kiwi-checkbox-menu';

describe('kiwi-checkbox-menu', () => {
  it('builds', async () => {
    const menu = await newSpecPage({
      components: [KiwiCheckboxMenu],
      html: `
        <kiwi-checkbox-menu></kiwi-checkbox-menu>
      `,
    });
    expect(menu.root).toBeTruthy();
  });

  it('renders', async () => {
    const menu = await newSpecPage({
      components: [KiwiCheckboxMenu],
      html: `
        <kiwi-checkbox-menu>
          <li>There can be only one</li>
        </kiwi-checkbox-menu>
      `,
    });

    expect(menu.root).toMatchSnapshot();
  });
});
