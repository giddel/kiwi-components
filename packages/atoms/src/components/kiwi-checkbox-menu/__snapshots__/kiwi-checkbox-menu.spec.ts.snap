// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-checkbox-menu renders 1`] = `
<kiwi-checkbox-menu>
  <!---->
  <ul class="checkbox-menu mb-0">
    <li>
      There can be only one
    </li>
  </ul>
</kiwi-checkbox-menu>
`;
